### ISSUE TITLE
Basic cross build support for C++ libraries and applications

### ISSUE THREAD

**Issue by adammurdoch** [gradle/gradle-native#12]
Allow a C++ application or library to be built from C++ source, and consumed from:

Another project in the same build
Another project in a different build in the same composite.
Another project in a different build, via a binary repository.
Assume a single platform (say, whatever the current platform is), a hard-coded set of tool chains and conventional source locations.

This epic de-risks our basic strategy for stable native support, which is to reuse the task and tool chain implementations from the software model plugins but with a new and much simpler DSL and model. It also unblocks initial Swift support by providing a pattern that can be reused and infrastructure that can be shared. It also unblocks native migrations to a large degree, as an existing build can be wrapped by Gradle and the dependency management capabilities of Gradle used to orchestrate.

Out-of-scope for this epic:

Pretty much any configurability.
Support for multiple variants - platforms, build types, etc.
Support for any languages other than C++.
Documentation, blog posts or any announcements.
Support for Ivy repositories or pre-built binaries.
Support for libraries that don't include both headers and runtime files, such as header only libraries or windows resource only libraries.
